### ISSUE TITLE
Update to Payara 5.2021.10 [5]

### ISSUE THREAD

**Issue by StefanKock** [hzi-braunschweig/SORMAS-Project#4601]
### Problem Description

- Currently used: Payara-5.2020.2 (5.194 for development) with some other dependencies (`glassfish/modules`) we want to update.
- paraya-5.2021.1 CE dropped the production template with we currently use for domain setup

### Proposed Change

1. Update to ~~`payara-5.2021.6`~~ `payara-5.2021.10`
2. Switch back to domain1 template (default) and configure relevant settings of production template.
3. Test whether we can drop 5.194 use for developers (if #2511 is mitigated) and debug mode with IntelliJ still works.
4. Check whether something needs to be done in `MultiAuthenticationMechanism` (method `logoutEvent`: _This is required since the payara-api version < 5.2020.3 doesn't have automated IDP logout built in it._)
5. Document in DEVELOPMENT_ENVIRONMENT.md (and link it in TROUBLESHOOTING.md) that because of redeployment problems a clean undeployment has to happen before stopping the domain to redeploy the artifacts.
6. Sync changes with SORMAS-Docker (see ticket https://github.com/hzi-braunschweig/SORMAS-Docker/issues/322 )

### Additional Information
- [Payara EE 5.25.0 production template documentation](https://docs.payara.fish/enterprise/docs/5.25.0/documentation/payara-server/production-ready-domain.html).
  - Production template did enforce `fish.payara.classloading.delegate=false` ([doc](https://docs.payara.fish/community/docs/5.2021.6/documentation/payara-server/classloading.html)).
- Increased to payara-5.2021.10 to cover more bugfixes and patched modules
- This ticket requires you to set up your payara domain freshly with the server-setup.sh and migrate all data and settings from the old domain: same database, same file paths, sormas.properties, logback.xml -> new ticket #7575 created.

**Comment by StefanKock** [comment]
In similar tech stacks, we migrated from payara-5.2020.3 to payara-5.2021.1 (and back to domain1 template) without noticable drawback, but the deployment problems why we use 5.194 for developers in this project was not visited.

**Comment by StefanKock** [comment]
> Test whether we can drop 5.194 use for developers

I quickly tested to redeploy sormas-ear (per Eclipse IDE) with a cloned `sormas` domain (not recreated per script) and `payara-5.2021.6` and it worked. I did not run into any problems. It seems we can avoid using an old Paraya.

**Comment by StefanKock** [comment]
I quickly updated the script and dependencies to Payara 5.2021.6. Unfortunately, we have breaking changes concerning OpenId Connect that need to fixed first before testing the domain setup (info for @vidi42).

**Comment by vidi42** [comment]
Fixed the OpenId Connect compilation issues, but cannot start the UI to test because of the `ConfigFacade`

I also just copied the payara5 from `5.2021.6` over the one from `5.194`

<details>
<summary>Stacktrace</summary>

```
[2021-09-08T13:05:10.697+0300] [Payara 5.2021.6] [SEVERE] [] [javax.enterprise.system.core] [tid: _ThreadID=116 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1631095510697] [levelValue: 1000] [[
          Exception while loading the app : javax.ejb.CreateException: Initialization failed for Singleton StartupShutdownService
        javax.ejb.CreateException: Initialization failed for Singleton StartupShutdownService
	        at com.sun.ejb.containers.AbstractSingletonContainer.createSingletonEJB(AbstractSingletonContainer.java:542)
	        at com.sun.ejb.containers.AbstractSingletonContainer.access$000(AbstractSingletonContainer.java:81)
	        at com.sun.ejb.containers.AbstractSingletonContainer$SingletonContextFactory.create(AbstractSingletonContainer.java:687)
	        at com.sun.ejb.containers.AbstractSingletonContainer.instantiateSingletonInstance(AbstractSingletonContainer.java:421)
	        at org.glassfish.ejb.startup.SingletonLifeCycleManager.initializeSingleton(SingletonLifeCycleManager.java:219)
	        at org.glassfish.ejb.startup.SingletonLifeCycleManager.initializeSingleton(SingletonLifeCycleManager.java:180)
	        at org.glassfish.ejb.startup.SingletonLifeCycleManager.doStartup(SingletonLifeCycleManager.java:159)
	        at org.glassfish.ejb.startup.EjbApplication.start(EjbApplication.java:171)
	        at org.glassfish.internal.data.EngineRef.start(EngineRef.java:123)
	        at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:293)
	        at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:364)
	        at com.sun.enterprise.v3.server.ApplicationLifecycle.initialize(ApplicationLifecycle.java:623)
	        at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:580)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:556)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:552)
	        at java.base/java.security.AccessController.doPrivileged(Native Method)
	        at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:551)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:582)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:574)
	        at java.base/java.security.AccessController.doPrivileged(Native Method)
	        at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:573)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1497)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:120)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1879)
	        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1755)
	        at com.sun.enterprise.v3.admin.AdminAdapter.doCommand(AdminAdapter.java:604)
	        at com.sun.enterprise.v3.admin.AdminAdapter.onMissingResource(AdminAdapter.java:256)
	        at org.glassfish.grizzly.http.server.StaticHttpHandlerBase.service(StaticHttpHandlerBase.java:166)
	        at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
	        at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
	        at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
	        at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
	        at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
	        at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	        at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	        at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	        at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	        at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
	        at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
	        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
	        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
	        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	        at java.base/java.lang.Thread.run(Thread.java:834)
        Caused by: javax.ejb.EJBException: javax.ejb.EJBException: javax.ejb.CreateException: Could not create stateless EJB
	        at com.sun.ejb.containers.StatelessSessionContainer._getContext(StatelessSessionContainer.java:399)
	        at com.sun.ejb.containers.BaseContainer.getContext(BaseContainer.java:2598)
	        at com.sun.ejb.containers.BaseContainer.preInvoke(BaseContainer.java:2019)
	        at com.sun.ejb.containers.EJBObjectInvocationHandler.invoke(EJBObjectInvocationHandler.java:204)
	        at com.sun.ejb.containers.EJBObjectInvocationHandlerDelegate.invoke(EJBObjectInvocationHandlerDelegate.java:79)
	        at com.sun.proxy.$Proxy2147.getCountryLocale(Unknown Source)
	        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	        at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:143)
	        at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:173)
	        at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:528)
	        at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:199)
	        at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1565)
	        at com.sun.corba.ee.impl.protocol.SharedCDRClientRequestDispatcherImpl.marshalingComplete(SharedCDRClientRequestDispatcherImpl.java:128)
	        at com.sun.corba.ee.impl.protocol.ClientDelegateImpl.invoke(ClientDelegateImpl.java:259)
	        at com.sun.corba.ee.impl.presentation.rmi.StubInvocationHandlerImpl.privateInvoke(StubInvocationHandlerImpl.java:189)
	        at com.sun.corba.ee.impl.presentation.rmi.StubInvocationHandlerImpl.invoke(StubInvocationHandlerImpl.java:140)
	        at com.sun.corba.ee.impl.presentation.rmi.codegen.CodegenStubBase.invoke(CodegenStubBase.java:226)
	        at de.symeda.sormas.api.__ConfigFacade_Remote_DynamicStub.getCountryLocale(de/symeda/sormas/api/__ConfigFacade_Remote_DynamicStub.java)
	        at de.symeda.sormas.api._ConfigFacade_Wrapper.getCountryLocale(de/symeda/sormas/api/_ConfigFacade_Wrapper.java)
	        at de.symeda.sormas.ui.StartupShutdownService.startup(StartupShutdownService.java:38)
	        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	        at com.sun.ejb.containers.interceptors.BeanCallbackInterceptor.intercept(InterceptorManager.java:1022)
	        at com.sun.ejb.containers.interceptors.CallbackChainImpl.invokeNext(CallbackChainImpl.java:72)
	        at com.sun.ejb.containers.interceptors.CallbackInvocationContext.proceed(CallbackInvocationContext.java:204)
	        at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.doCall(SystemInterceptorProxy.java:163)
	        at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.init(SystemInterceptorProxy.java:125)
	        at jdk.internal.reflect.GeneratedMethodAccessor282.invoke(Unknown Source)
	        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	        at com.sun.ejb.containers.interceptors.CallbackInterceptor.intercept(InterceptorManager.java:978)
	        at com.sun.ejb.containers.interceptors.CallbackChainImpl.invokeNext(CallbackChainImpl.java:72)
	        at com.sun.ejb.containers.interceptors.CallbackInvocationContext.proceed(CallbackInvocationContext.java:204)
	        at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:81)
	        at org.jboss.weld.module.ejb.SessionBeanInterceptor.aroundInvoke(SessionBeanInterceptor.java:52)
	        at jdk.internal.reflect.GeneratedMethodAccessor141.invoke(Unknown Source)
	        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	        at com.sun.ejb.containers.interceptors.CallbackInterceptor.intercept(InterceptorManager.java:978)
	        at com.sun.ejb.containers.interceptors.CallbackChainImpl.invokeNext(CallbackChainImpl.java:72)
	        at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:418)
	        at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:381)
	        at com.sun.ejb.containers.BaseContainer.intercept(BaseContainer.java:2062)
	        at com.sun.ejb.containers.AbstractSingletonContainer.createSingletonEJB(AbstractSingletonContainer.java:529)
	        ... 48 more
        Caused by: javax.ejb.EJBException: javax.ejb.CreateException: Could not create stateless EJB
	        at com.sun.ejb.containers.StatelessSessionContainer$SessionContextFactory.create(StatelessSessionContainer.java:653)
	        at com.sun.ejb.containers.util.pool.NonBlockingPool.getObject(NonBlockingPool.java:219)
	        at com.sun.ejb.containers.StatelessSessionContainer._getContext(StatelessSessionContainer.java:395)
	        ... 96 more
        Caused by: javax.ejb.CreateException: Could not create stateless EJB
	        at com.sun.ejb.containers.StatelessSessionContainer.createStatelessEJB(StatelessSessionContainer.java:473)
	        at com.sun.ejb.containers.StatelessSessionContainer.access$000(StatelessSessionContainer.java:100)
	        at com.sun.ejb.containers.StatelessSessionContainer$SessionContextFactory.create(StatelessSessionContainer.java:651)
	        ... 98 more
        Caused by: java.lang.IllegalStateException: Exception attempting to inject Env-Prop: de.symeda.sormas.backend.common.ConfigFacadeEjb/props@Field-Injectable Resource. Class name = de.symeda.sormas.backend.common.ConfigFacadeEjb Field name=props@java.lang.String@@@ into class de.symeda.sormas.backend.common.ConfigFacadeEjb: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	        at org.glassfish.weld.services.InjectionServicesImpl.aroundInject(InjectionServicesImpl.java:194)
	        at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:46)
	        at org.jboss.weld.injection.producer.DefaultInjector.inject(DefaultInjector.java:73)
	        at org.jboss.weld.module.ejb.DynamicInjectionPointInjector.inject(DynamicInjectionPointInjector.java:61)
	        at org.jboss.weld.module.ejb.SessionBeanInjectionTarget.inject(SessionBeanInjectionTarget.java:138)
	        at org.glassfish.weld.services.JCDIServiceImpl$JCDIInjectionContextImpl.inject(JCDIServiceImpl.java:576)
	        at org.glassfish.weld.services.JCDIServiceImpl.injectEJBInstance(JCDIServiceImpl.java:323)
	        at com.sun.ejb.containers.BaseContainer.injectEjbInstance(BaseContainer.java:1813)
	        at com.sun.ejb.containers.StatelessSessionContainer.createStatelessEJB(StatelessSessionContainer.java:437)
	        ... 100 more
        Caused by: com.sun.enterprise.container.common.spi.util.InjectionException: Exception attempting to inject Env-Prop: de.symeda.sormas.backend.common.ConfigFacadeEjb/props@Field-Injectable Resource. Class name = de.symeda.sormas.backend.common.ConfigFacadeEjb Field name=props@java.lang.String@@@ into class de.symeda.sormas.backend.common.ConfigFacadeEjb: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:638)
	        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.inject(InjectionManagerImpl.java:439)
	        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.injectInstance(InjectionManagerImpl.java:153)
	        at org.glassfish.weld.services.InjectionServicesImpl.aroundInject(InjectionServicesImpl.java:184)
	        ... 108 more
        Caused by: java.lang.IllegalArgumentException: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	        at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
	        at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)
	        at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.ensureObj(UnsafeFieldAccessorImpl.java:58)
	        at java.base/jdk.internal.reflect.UnsafeObjectFieldAccessorImpl.set(UnsafeObjectFieldAccessorImpl.java:75)
	        at java.base/java.lang.reflect.Field.set(Field.java:780)
	        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:594)
	        ... 111 more
        ]]
```
</details>

@StefanKock maybe you can have a look

**Comment by StefanKock** [comment]
I have problems properly resolving `org.glassfish.soteria:javax.security.enterprise:1.1-b01.payara-p4`but deployment of ear and ui works on my cloned domain with payara 5.2021.6

**Comment by vidi42** [comment]
> I have problems properly resolving `org.glassfish.soteria:javax.security.enterprise:1.1-b01.payara-p4`but deployment of ear and ui works on my cloned domain with payara 5.2021.6

Didn't had this issue locally. Can you add the following repository and check if it solves the issue for you?

```
<repository>
    <id>payara-patched-externals</id>
    <name>Payara Patched Externals</name>
    <url>https://raw.github.com/payara/Payara_PatchedProjects/master</url>
</repository>
```

**Comment by StefanKock** [comment]
I added the repo to the Nexus I'm connected and tried to call the URL, alsways a 400 error. https://github.com/payara/Payara_PatchedProjects is readonly, maybe therefore it is not reachable anymore ...

**Comment by vidi42** [comment]
Worked for me. I just committed an updated `pom.xml`.

```
[INFO] --------------------< de.symeda.sormas:sormas-rest >--------------------
[INFO] Building sormas-rest 1.64.0-SNAPSHOT                               [6/9]
[INFO] --------------------------------[ war ]---------------------------------
Downloading from central: https://jcenter.bintray.com/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.pom
Downloading from vaadin-addons: https://maven.vaadin.com/vaadin-addons/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.pom
Downloading from Novatec: https://repository.novatec-gmbh.de/content/repositories/novatec/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.pom
Downloading from osgeo: https://repo.osgeo.org/repository/release/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.pom
Downloading from payara-patched-externals: https://raw.github.com/payara/Payara_PatchedProjects/master/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.pom
Downloaded from payara-patched-externals: https://raw.github.com/payara/Payara_PatchedProjects/master/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.pom (8.1 kB at 7.3 k
B/s)
Downloading from central: https://jcenter.bintray.com/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.jar
Downloading from vaadin-addons: https://maven.vaadin.com/vaadin-addons/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.jar
Downloading from Novatec: https://repository.novatec-gmbh.de/content/repositories/novatec/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.jar
Downloading from osgeo: https://repo.osgeo.org/repository/release/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.jar
Downloading from payara-patched-externals: https://raw.github.com/payara/Payara_PatchedProjects/master/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.jar
Downloaded from payara-patched-externals: https://raw.github.com/payara/Payara_PatchedProjects/master/org/glassfish/soteria/javax.security.enterprise/1.1-b01.payara-p4/javax.security.enterprise-1.1-b01.payara-p4.jar (133 kB at 123 k
B/s)
```

**Comment by StefanKock** [comment]
Current status:  

1. The redeployment of `sormas-ear` causes the access from outside to fail because of mismatching classes from different `EarClassLoader` instances (debug break point on `UnsafeFieldAccessorImpl.java:58` to see this). 
<details>
<summary>StackTrace when (re-)deploying sormas-ui after 2nd deployment of sormas-ear</summary>

```
2021-09-09T16:31:44.842+0200|SCHWERWIEGEND: Exception while loading the app : javax.ejb.CreateException: Initialization failed for Singleton StartupShutdownService
javax.ejb.CreateException: Initialization failed for Singleton StartupShutdownService
	at com.sun.ejb.containers.AbstractSingletonContainer.createSingletonEJB(AbstractSingletonContainer.java:542)
	at com.sun.ejb.containers.AbstractSingletonContainer.access$000(AbstractSingletonContainer.java:81)
	at com.sun.ejb.containers.AbstractSingletonContainer$SingletonContextFactory.create(AbstractSingletonContainer.java:687)
	at com.sun.ejb.containers.AbstractSingletonContainer.instantiateSingletonInstance(AbstractSingletonContainer.java:421)
	at org.glassfish.ejb.startup.SingletonLifeCycleManager.initializeSingleton(SingletonLifeCycleManager.java:219)
	at org.glassfish.ejb.startup.SingletonLifeCycleManager.initializeSingleton(SingletonLifeCycleManager.java:180)
	at org.glassfish.ejb.startup.SingletonLifeCycleManager.doStartup(SingletonLifeCycleManager.java:159)
	at org.glassfish.ejb.startup.EjbApplication.start(EjbApplication.java:171)
	at org.glassfish.internal.data.EngineRef.start(EngineRef.java:123)
	at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:293)
	at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:364)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.initialize(ApplicationLifecycle.java:623)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:580)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:556)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:552)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:551)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:582)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:574)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:573)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1497)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:120)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1879)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1755)
	at com.sun.enterprise.v3.admin.AdminAdapter.doCommand(AdminAdapter.java:604)
	at com.sun.enterprise.v3.admin.AdminAdapter.onMissingResource(AdminAdapter.java:256)
	at org.glassfish.grizzly.http.server.StaticHttpHandlerBase.service(StaticHttpHandlerBase.java:166)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.ejb.EJBException: javax.ejb.EJBException: javax.ejb.CreateException: Could not create stateless EJB
	at com.sun.ejb.containers.StatelessSessionContainer._getContext(StatelessSessionContainer.java:399)
	at com.sun.ejb.containers.BaseContainer.getContext(BaseContainer.java:2598)
	at com.sun.ejb.containers.BaseContainer.preInvoke(BaseContainer.java:2019)
	at com.sun.ejb.containers.EJBObjectInvocationHandler.invoke(EJBObjectInvocationHandler.java:204)
	at com.sun.ejb.containers.EJBObjectInvocationHandlerDelegate.invoke(EJBObjectInvocationHandlerDelegate.java:79)
	at com.sun.proxy.$Proxy1371.getCountryLocale(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:143)
	at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:173)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:528)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:199)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1565)
	at com.sun.corba.ee.impl.protocol.SharedCDRClientRequestDispatcherImpl.marshalingComplete(SharedCDRClientRequestDispatcherImpl.java:128)
	at com.sun.corba.ee.impl.protocol.ClientDelegateImpl.invoke(ClientDelegateImpl.java:259)
	at com.sun.corba.ee.impl.presentation.rmi.StubInvocationHandlerImpl.privateInvoke(StubInvocationHandlerImpl.java:189)
	at com.sun.corba.ee.impl.presentation.rmi.StubInvocationHandlerImpl.invoke(StubInvocationHandlerImpl.java:140)
	at com.sun.corba.ee.impl.presentation.rmi.codegen.CodegenStubBase.invoke(CodegenStubBase.java:226)
	at de.symeda.sormas.api.__ConfigFacade_Remote_DynamicStub.getCountryLocale(de/symeda/sormas/api/__ConfigFacade_Remote_DynamicStub.java)
	at de.symeda.sormas.api._ConfigFacade_Wrapper.getCountryLocale(de/symeda/sormas/api/_ConfigFacade_Wrapper.java)
	at de.symeda.sormas.ui.StartupShutdownService.startup(StartupShutdownService.java:38)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.sun.ejb.containers.interceptors.BeanCallbackInterceptor.intercept(InterceptorManager.java:1022)
	at com.sun.ejb.containers.interceptors.CallbackChainImpl.invokeNext(CallbackChainImpl.java:72)
	at com.sun.ejb.containers.interceptors.CallbackInvocationContext.proceed(CallbackInvocationContext.java:204)
	at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.doCall(SystemInterceptorProxy.java:163)
	at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.init(SystemInterceptorProxy.java:125)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.sun.ejb.containers.interceptors.CallbackInterceptor.intercept(InterceptorManager.java:978)
	at com.sun.ejb.containers.interceptors.CallbackChainImpl.invokeNext(CallbackChainImpl.java:72)
	at com.sun.ejb.containers.interceptors.CallbackInvocationContext.proceed(CallbackInvocationContext.java:204)
	at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:81)
	at org.jboss.weld.module.ejb.SessionBeanInterceptor.aroundInvoke(SessionBeanInterceptor.java:52)
	at jdk.internal.reflect.GeneratedMethodAccessor620.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.sun.ejb.containers.interceptors.CallbackInterceptor.intercept(InterceptorManager.java:978)
	at com.sun.ejb.containers.interceptors.CallbackChainImpl.invokeNext(CallbackChainImpl.java:72)
	at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:418)
	at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:381)
	at com.sun.ejb.containers.BaseContainer.intercept(BaseContainer.java:2062)
	at com.sun.ejb.containers.AbstractSingletonContainer.createSingletonEJB(AbstractSingletonContainer.java:529)
	... 48 more
Caused by: javax.ejb.EJBException: javax.ejb.CreateException: Could not create stateless EJB
	at com.sun.ejb.containers.StatelessSessionContainer$SessionContextFactory.create(StatelessSessionContainer.java:653)
	at com.sun.ejb.containers.util.pool.NonBlockingPool.getObject(NonBlockingPool.java:219)
	at com.sun.ejb.containers.StatelessSessionContainer._getContext(StatelessSessionContainer.java:395)
	... 97 more
Caused by: javax.ejb.CreateException: Could not create stateless EJB
	at com.sun.ejb.containers.StatelessSessionContainer.createStatelessEJB(StatelessSessionContainer.java:473)
	at com.sun.ejb.containers.StatelessSessionContainer.access$000(StatelessSessionContainer.java:100)
	at com.sun.ejb.containers.StatelessSessionContainer$SessionContextFactory.create(StatelessSessionContainer.java:651)
	... 99 more
Caused by: java.lang.IllegalStateException: Exception attempting to inject Env-Prop: de.symeda.sormas.backend.common.ConfigFacadeEjb/props@Field-Injectable Resource. Class name = de.symeda.sormas.backend.common.ConfigFacadeEjb Field name=props@java.lang.String@@@ into class de.symeda.sormas.backend.common.ConfigFacadeEjb: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	at org.glassfish.weld.services.InjectionServicesImpl.aroundInject(InjectionServicesImpl.java:194)
	at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:46)
	at org.jboss.weld.injection.producer.DefaultInjector.inject(DefaultInjector.java:73)
	at org.jboss.weld.module.ejb.DynamicInjectionPointInjector.inject(DynamicInjectionPointInjector.java:61)
	at org.jboss.weld.module.ejb.SessionBeanInjectionTarget.inject(SessionBeanInjectionTarget.java:138)
	at org.glassfish.weld.services.JCDIServiceImpl$JCDIInjectionContextImpl.inject(JCDIServiceImpl.java:576)
	at org.glassfish.weld.services.JCDIServiceImpl.injectEJBInstance(JCDIServiceImpl.java:323)
	at com.sun.ejb.containers.BaseContainer.injectEjbInstance(BaseContainer.java:1813)
	at com.sun.ejb.containers.StatelessSessionContainer.createStatelessEJB(StatelessSessionContainer.java:437)
	... 101 more
Caused by: com.sun.enterprise.container.common.spi.util.InjectionException: Exception attempting to inject Env-Prop: de.symeda.sormas.backend.common.ConfigFacadeEjb/props@Field-Injectable Resource. Class name = de.symeda.sormas.backend.common.ConfigFacadeEjb Field name=props@java.lang.String@@@ into class de.symeda.sormas.backend.common.ConfigFacadeEjb: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:638)
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.inject(InjectionManagerImpl.java:439)
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.injectInstance(InjectionManagerImpl.java:153)
	at org.glassfish.weld.services.InjectionServicesImpl.aroundInject(InjectionServicesImpl.java:184)
	... 109 more
Caused by: java.lang.IllegalArgumentException: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.ensureObj(UnsafeFieldAccessorImpl.java:58)
	at java.base/jdk.internal.reflect.UnsafeObjectFieldAccessorImpl.set(UnsafeObjectFieldAccessorImpl.java:75)
	at java.base/java.lang.reflect.Field.set(Field.java:780)
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:594)
	... 112 more
```

</details>

2. I noticed this message: `No valid EE environment for injection of org.glassfish.cdi.transaction.TransactionScopedCDIEventHelperImpl` around the auditlog. But I tested, at least `AuditlogEntry`s were created.

<details>
<summary>Log messages around "No valid EE environment"</summary>

```
2021-09-09T11:39:23.666+0200|INFORMATION: D113923.665|     > Finished in 1 ms: AuditContextProducer.provideAuditor (de.symeda.sormas.backend.util.PerformanceLoggingInterceptor.java:51)

2021-09-09T11:39:23.666+0200|SCHWERWIEGEND: No valid EE environment for injection of org.glassfish.cdi.transaction.TransactionScopedCDIEventHelperImpl
2021-09-09T11:39:23.667+0200|INFORMATION: T113923.667|     > Started: AuditContextProducer.getCurrentlyLoggedInUser with parameters 'null' (de.symeda.sormas.backend.util.PerformanceLoggingInterceptor.java:42)
```

</details>

3. WebappClassLoader of `sormas-ui` fails to be removed.

<details>
<summary>Log message</summary>

```
2021-09-09T11:36:18.430+0200|SCHWERWIEGEND: The web application [/sormas-ui] created a ThreadLocal with key of type [org.glassfish.pfl.dynamic.codegen.impl.CurrentClassLoader$1] (value [org.glassfish.pfl.dynamic.codegen.impl.CurrentClassLoader$1@5767e78f]) and a value of type [org.glassfish.web.loader.WebappClassLoader] (value [WebappClassLoader (delegate=true; repositoryURLs=file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/gentyref.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/keycloak-common.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-compatibility-shared.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/flute.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/sac.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/classes/,file:/C:/srv/payara-domains/sormas_clone/generated/ejb/sormas-ui/,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-server.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/aopalliance.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/spring-core.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/keycloak-core.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/sormas-api.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-compatibility-server.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/extended-token-field.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-sass-compiler.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/spring-context.jar,file:/C:/srv/payara-domains/sormas_clone/generated/sormas-ui/,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/popupbutton.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-shared.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/sormas-widgetset.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/spring-beans.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/spring-expression.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/spring-aop.jar,file:/C:/srv/payara-domains/sormas_clone/eclipseApps/sormas-ui/WEB-INF/lib/wcslib-vaadin-widget-multifileupload.jar) JarNames: []; canonicalLoaderDir: C:\srv\payara-domains\sormas_clone\generated\jsp\sormas-ui\loader_2014963651\; Object: 7819e7c3 Created: 09.09.2021, 11:36:05]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
```

</summary>

**Comment by StefanKock** [comment]
I was investigating how to get rid of leftover class loaders: 
- https://java.jiderhamn.se/2012/03/04/classloader-leaks-vi-this-means-war-leak-prevention-library/
- https://github.com/mjiderhamn/classloader-leak-prevention

I don't know how to invoke this in an EAR setup, and didn't yet try to integrate it into the WAR files to see if it solves the class instance mismatch problem.

**Comment by vidi42** [comment]
Tried adding the [classloader-leak-prevention](https://github.com/mjiderhamn/classloader-leak-prevention) library to the pom.xml of the `sormas-ear` file and that seemed to fix the issue in the beginning as the redeploy worked for a couple of times, however this was just a coincidence as things broke again afterwards.

Tried with adding the library to the `sormas-ui` module as well, but no luck.

Putting this on hold for now as I don't have any new leads to go further on.

My assumption is the redeploy issue was introduced when the [Hot Deploy](https://blog.payara.fish/hot-deploy-feature-in-payara-platform-5.201) option was introduced in Payara.

**Comment by StefanKock** [comment]
I took the changes that #6939 made to avoid architecture breach when using `FacadeProvider`, but it did not solve the redeploy problem  addressed in https://github.com/hzi-braunschweig/SORMAS-Project/issues/4601#issuecomment-916167447 .

**Comment by StefanKock** [comment]
Discussed with @vidi42: We are pushing forward to payara-5.2021.10 (latest release, see https://docs.payara.fish/community/docs/release-notes/release-notes-history.html) and retest.

**Comment by StefanKock** [comment]
I tested again the deployment with a fresh set up payara domain. The redeployment problem is still present: Once you undeploy `sormas-ear.ear`, after the 2nd deployment in the running payara domain the deployment of `sormas-ui.war` and `sormas-rest.war` fails with such an exception:

<details>
<summary>Exception / Classloader logging</summary>

```
Caused by: com.sun.enterprise.container.common.spi.util.InjectionException: Exception attempting to inject Env-Prop: de.symeda.sormas.backend.common.ConfigFacadeEjb/props@Field-Injectable Resource. Class name = de.symeda.sormas.backend.common.ConfigFacadeEjb Field name=props@java.lang.String@@@ into class de.symeda.sormas.backend.common.ConfigFacadeEjb: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:638)
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.inject(InjectionManagerImpl.java:439)
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.injectInstance(InjectionManagerImpl.java:153)
	at org.glassfish.weld.services.InjectionServicesImpl.aroundInject(InjectionServicesImpl.java:184)
	... 108 more
Caused by: java.lang.IllegalArgumentException: Can not set java.util.Properties field de.symeda.sormas.backend.common.ConfigFacadeEjb.props to de.symeda.sormas.backend.common.ConfigFacadeEjb
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.ensureObj(UnsafeFieldAccessorImpl.java:58)
	at java.base/jdk.internal.reflect.UnsafeObjectFieldAccessorImpl.set(UnsafeObjectFieldAccessorImpl.java:75)
	at java.base/java.lang.reflect.Field.set(Field.java:780)
	at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:594)
	... 111 more
```

It happens when deploying via Eclipse IDE Payara Tools. If you keep `sormas-ear.ear` in place, you can redeploy the WARs as often as you want without problems.

This message appears when undeploying `sormas-ui` and `sormas-app`:
```
2021-12-22T13:25:46.884+0100|SCHWERWIEGEND: The web application [/sormas-ui] created a ThreadLocal with key of type [org.glassfish.pfl.dynamic.codegen.impl.CurrentClassLoader$1] (value [org.glassfish.pfl.dynamic.codegen.impl.CurrentClassLoader$1@4c111195]) and a value of type [org.glassfish.web.loader.WebappClassLoader] (value [WebappClassLoader (delegate=true; repositoryURLs=file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/flute.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-shared.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/popupbutton.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/spring-expression.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/keycloak-core.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/spring-beans.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-sass-compiler.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-compatibility-shared.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/keycloak-common.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/spring-aop.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/spring-core.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/sormas-widgetset.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-compatibility-server.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/extended-token-field.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/vaadin-server.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/wcslib-vaadin-widget-multifileupload.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/spring-context.jar,file:/C:/srv/payara-domains/sormas2/generated/sormas-ui/,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/gentyref.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/classes/,file:/C:/srv/payara-domains/sormas2/generated/ejb/sormas-ui/,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/sormas-api.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-ui/WEB-INF/lib/sac.jar) JarNames: []; canonicalLoaderDir: C:\srv\payara-domains\sormas2\generated\jsp\sormas-ui\loader_2028572776\; Object: 78e99068 Created: 22.12.2021, 13:23:06]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.

2021-12-22T13:25:47.723+0100|SCHWERWIEGEND: The web application [/sormas-rest] created a ThreadLocal with key of type [org.glassfish.pfl.dynamic.codegen.impl.CurrentClassLoader$1] (value [org.glassfish.pfl.dynamic.codegen.impl.CurrentClassLoader$1@4c111195]) and a value of type [org.glassfish.web.loader.WebappClassLoader] (value [WebappClassLoader (delegate=true; repositoryURLs=file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/spring-aop.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/swagger-core.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/swagger-annotations.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/keycloak-servlet-filter-adapter.jar,file:/C:/srv/payara-domains/sormas2/generated/ejb/sormas-rest/,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/spring-context.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/swagger-models.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/classgraph.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/sormas-api.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/swagger-jaxrs2.jar,file:/C:/srv/payara-domains/sormas2/generated/sormas-rest/,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/swagger-integration.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/keycloak-adapter-core.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/spring-expression.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/spring-beans.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/swagger-jaxrs2-servlet-initializer-v2.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/keycloak-core.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/jackson-datatype-jsr310.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/keycloak-common.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/spring-core.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/classes/,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/keycloak-servlet-adapter-spi.jar,file:/C:/srv/payara-domains/sormas2/eclipseApps/sormas-rest/WEB-INF/lib/keycloak-adapter-spi.jar) JarNames: []; canonicalLoaderDir: C:\srv\payara-domains\sormas2\generated\jsp\sormas-rest\loader_1718815894\; Object: 66730c96 Created: 22.12.2021, 13:22:16]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
```

</details>

**Comment by StefanKock** [comment]
Code changes are ready to review, but SORMAS-Docker needs to be adjusted in sync.

**Comment by vidi42** [comment]
We should check out https://github.com/payara/Payara/pull/5394 and see if it's an alternative to re-deploy

**Comment by StefanKock** [comment]
I'm currently struggling to start the payara domain with 5.2021.10 (but it worked on 2021-12-22).
